### ISSUE TITLE
plain "go get" doesn't work on the repository

### ISSUE THREAD

**Issue by danstender** [digitalocean/doctl#42]
`$ go get github.com/digitalocean/doctl/cmd/doctl` doesn't work out of the box:

```
$ go get github.com/digitalocean/doctl/cmd/doctl
# cd /home/user/mygo/src/github.com/digitalocean/doctl; git submodule update --init --recursive
fatal: no submodule mapping found in .gitmodules for path 'vendor/github.com/inconshreveable/mousetrap'
package github.com/digitalocean/doctl/cmd/doctl: exit status 128
```

DS

**Comment by jungle-boogie** [comment]
Yep, I experienced this and thought it was something in my setup. 

I can't use the binaries as there's no version for freebsd 32/64bit.

**Comment by jakebasile** [comment]
Ran into this too. Looks like when #1 copied everything in, a few deps were missed. `go get` gets confused and complains.

I fixed it in jakebasile/doctl@aa6995516140ddbc143eb50473aaf3cd32332f1c by checking out the referenced commits and it appears to work when I try `go get github.com/jakebasile/doctl`, but it will not work with `-u` as it fails to get the correct paths from my project.

Not sure if I did the vendoring the preferred way, but it seems to work. Let me know if you'd like a PR, or maybe there's a better way to fix this.

**Comment by bryanl** [comment]
You should be able to do `go get github.com/digitalocean/doctl/...`. Currently, one of the vendored repos has a weird submodule thing I need to explore. The good news is that even with the `error 1`, you can still install a binary with `go install github.com/digitalocean/doctl/cmd/doctl`. I'll work on tracking down that submodule issue.

**Comment by macktic** [comment]
This issue seems to still be open, I have to install from source as there is no arm version.
downloading the source and using `go build doctl/cmd/doctl` works fine.

**Comment by nesv** [comment]
Just to tack onto this issue, I'm getting the following error:

```
$ go get -v -u github.com/digitalocean/doctl/...
github.com/digitalocean/doctl (download)
# cd /home/nesv/src/github.com/digitalocean/doctl; git submodule update --init --recursive
fatal: no submodule mapping found in .gitmodules for path 'vendor/github.com/vektra/errors'
package github.com/digitalocean/doctl: exit status 128
```

@bryanl: Is this the "weird submodule thing" you referred to?

**Comment by klauern** [comment]
@nesv Yes, I got that too.  It works when you follow the [`go install` instructions](https://github.com/digitalocean/doctl/issues/42#issuecomment-209125581) that @bryanl mentioned above.

**Comment by nesv** [comment]
@klauern Yep! I was just adding the error I received in hopes that it would help. :)

**Comment by claudioandre** [comment]
There are special entries in the index indicating submodules. Inside doctl dir, do:

```
$ git ls-files --stage | grep 160000
160000 c64d83aba85aa4392895aadeefabbd24e89f3580 0	vendor/github.com/vektra/errors
160000 47a058062601e235731714f89d7ec95c1ec47a5e 0	vendor/github.com/vektra/mockery
160000 f3a63969dd29f8cfa913fdaea22f30c1ed537cb3 0	vendor/golang.org/x/tools
```

If that special entry path isn't referenced in the .gitmodule, then I guess DigitalOcean needs to remove it (one can add it again later).
****
After the removal, I was able to build using ```go get github.com/MyFork```). And voila:
```
$ ls -l temp/bin/
total 15608
-rwxrwxr-x 1 me me 15979048 Ago 29 16:33 doctl
```

**Comment by dch** [comment]
@claudioandre  any chance of you sharing a few more steps on how you managed this? specifically how to remove this entry path, and what command you needed to generate the binary?

**Comment by dch** [comment]
I worked out the incantation to do this, see https://github.com/digitalocean/doctl/issues/47#issuecomment-278467149 for details.

**Comment by xmudrii** [comment]
@mauricio i think this one should be fixed as well with #201
